components/autofill/src/sync/credit_card/outgoing.rs

```diff
@@ -34,10 +49,15 @@ impl ProcessOutgoingRecordImpl for OutgoingCreditCardsImpl {
             FROM credit_cards_data l
             LEFT JOIN credit_cards_mirror m
             ON l.guid = m.guid
-            WHERE sync_change_counter > 0
-                OR l.guid NOT IN (
-                    SELECT m.guid
-                    FROM credit_cards_mirror m
+            WHERE
```

Comment:
This change prevents the integration I added from failing. However, it's probably preferable to prevent records with empty string encrypted numbers from being added in iOS.